app/controllers/dashboard/dashboard-activity-controller.js

```diff
@@ -13,13 +13,59 @@ const CORRELATION_HEADER = require('../../utils/correlation_header').CORRELATION
 const ConnectorClient = require('../../services/clients/connector_client').ConnectorClient
 const { isADirectDebitAccount } = require('../../services/clients/direct_debit_connector_client.js')
 const auth = require('../../services/auth_service.js')
-const connectorClient = () => new ConnectorClient(process.env.CONNECTOR_URL)
 const { datetime } = require('@govuk-pay/pay-js-commons').nunjucksFilters
+const goLiveStage = require('../../models/go-live-stage')
+
+const connectorClient = () => new ConnectorClient(process.env.CONNECTOR_URL)
 const getTimespanDays = (fromDateTime, toDateTime) => moment(toDateTime).diff(moment(fromDateTime), 'days')
 
 // Constants
 const clsXrayConfig = require('../../../config/xray-cls')
 
+const getLinksToDisplay = (service, account) => {
+  const links = ['manageService']
+
+  if (account.payment_provider === 'sandbox') {
+    links.push('demoPayment')
+    links.push('testPaymentLink')
+  } else if (account.paymentMethod === 'direct debit') {
+    links.push('directDebitPaymentFlow')
+  } else {
+    links.push('paymentLinks')
+  }
+
+  if (account.type === 'test') {
+    links.push('goLive')
+  }
+
+  return links
+}
+
+/**
+ * Gets an array of the css classes to be applied to the link boxes on the dashboard in the order they are displayed in.
+ * @param numberOfLinks the number of links that will be displayed
+ * @returns {Array}
+ */
+const getLinkBoxClasses = (numberOfLinks) => {
+  const columnClass = numberOfLinks % 3 === 0 ? 'flex-grid--column-third' : 'flex-grid--column-half'
```

Comment:
In general it would be good if this logic could be moved over to the template, [nunjucks](https://mozilla.github.io/nunjucks/) lets you do a fair amount of processing. 

I don't have a suggestion here and I don't think it should block the PR -- having a clean separation between the controller figuring out what to display and the templates deciding how to display it is useful for reasoning about how the service works.

Reply:
Yes I think that's sensible. Doing this here doesn't feel quite right, as the controller probably shouldn't be specifying the CSS classes, right?
Do you know if I would be able to do it in a similar way in Nunjucks? I think the most difficult part is deciding whether or not to display the border, deciding which column class to use would be straightforward enough in Nunjucks.

Reply:
I have got rid of using the bottom-border css class, and we're now using an nth-child selector in the css to add in the grey line separators to the links. Thanks for all the suggestions and help.